src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -2501,29 +2521,43 @@ public long getLastKeepAlive() {
     }
 
     /**
-     * Gets the packet to add the player from the tab-list.
+     * Gets the packet to render all the players that are online.
      *
-     * @return a {@link PlayerInfoPacket} to add the player
+     * @return a {@link PlayerInfoPacket} containing all online players
      */
-    @NotNull
-    protected PlayerInfoPacket getAddPlayerToList() {
+    protected @NotNull PlayerInfoPacket getAddAllPlayerToList() {
         PlayerInfoPacket playerInfoPacket = new PlayerInfoPacket(PlayerInfoPacket.Action.ADD_PLAYER);
+        for (Player player : MinecraftServer.getConnectionManager().getOnlinePlayers()) {
+            PlayerInfoPacket.AddPlayer addPlayer =
+                    new PlayerInfoPacket.AddPlayer(player.getUuid(), player.getUsername(), player.getGameMode(), player.getLatency());
+            addPlayer.displayName = player.getDisplayName();
+
+            // Skin support
+            if (player.getSkin() != null) {
+                final String textures = player.getSkin().getTextures();
+                final String signature = player.getSkin().getSignature();
+
+                PlayerInfoPacket.AddPlayer.Property prop =
+                        new PlayerInfoPacket.AddPlayer.Property("textures", textures, signature);
+                addPlayer.properties.add(prop);
+            }
 
-        PlayerInfoPacket.AddPlayer addPlayer =
-                new PlayerInfoPacket.AddPlayer(getUuid(), getUsername(), getGameMode(), getLatency());
-        addPlayer.displayName = displayName;
-
-        // Skin support
-        if (skin != null) {
-            final String textures = skin.getTextures();
-            final String signature = skin.getSignature();
-
-            PlayerInfoPacket.AddPlayer.Property prop =
-                    new PlayerInfoPacket.AddPlayer.Property("textures", textures, signature);
-            addPlayer.properties.add(prop);
+            playerInfoPacket.playerInfos.add(addPlayer);
         }
+        return playerInfoPacket;
+    }
 
-        playerInfoPacket.playerInfos.add(addPlayer);
+    /**
+     * Gets the packet to remove all the players from tablist.
+     *
+     * @return a {@link PlayerInfoPacket} to remove the players
+     */
+    protected @NotNull PlayerInfoPacket getRemoveAllPlayerToList() {
+        PlayerInfoPacket playerInfoPacket = new PlayerInfoPacket(PlayerInfoPacket.Action.REMOVE_PLAYER);
+        for (Player player : MinecraftServer.getConnectionManager().getOnlinePlayers()) {
```

Comment:
Not a fan of all player loop

Reply:
It's the only way for player skins to be cached.

Reply:
This only gets called in the init, to add and then remove all players to tablist, I'll consolidate them into one for loop.